### PR TITLE
Fixes NaNs in DataFrama item_id column issue #291

### PR DESCRIPTION
**Pull request by AaronSpieler**
Also the 'item_id' will be the new index.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/gluonts/dataset/common.py [hunk 1]
```diff
@@ -215,11 +215,12 @@ def __init__(
 
     def __iter__(self) -> Iterator[DataEntry]:
         for path in self.files():
-            for line in jsonl.JsonLinesFile(path):
+            for i, line in enumerate(jsonl.JsonLinesFile(path)):
                 data = self.process(line.content)
                 data["source"] = SourceContext(
-                    source=line.span, row=line.span.line
+                    source=line.span.path, row=line.span.line
```
Comment (AaronSpieler) [inline src/gluonts/dataset/common.py]: I think I accidentally leaked this from latest master branch, however, should be no problem when merging with master.

**Comment by kashif** [comment]
@AaronSpieler it might be best to check if `data[item_id]` is empty before overwriting it with an index

**Comment by vafl** [comment]
I think the index of the dataframe is already the row number, so we don't need to add it explicitly. 

The purpose of `item_id` was a bit different. It was supposed to be a unique id that the user can provide in the dataset with each time series, and which can then be used later for instance to join results with other information that the user has. We should probably update the documentation on this.

Was there a problem with the NaN column or is this more of a usability question?

**Comment by kashif** [comment]
@vafl yes more of a usability issue since my data has unique `item_id`s and I its handy to see the metrics for specific `item_ids`...

**Comment by vafl** [comment]
@kashif yes exactly. So I think we either do what you suggest and overwrite it only if it is not there. But perhaps it's best to keep the current behavior as is and update the docs instead.

**Comment by lostella** [comment]
Note that the original issue #291 is really about the "item_id" field missing in m4_hourly (and maybe other provided datasets). So another option is to make sure that such field is filled-in in the scripts that generate the datasets.

**Comment by lostella** [comment]
I think this can be closed in favor of a new one that adjusts the provided datasets, see my [comment](https://github.com/awslabs/gluon-ts/issues/291#issuecomment-576018487) in #291

**Pull request closed without merging**